hub/policy/src/test/java/uk/gov/ida/hub/policy/controllogic/TransitionStoreTest.java

```diff
@@ -0,0 +1,510 @@
+    }
+
+    @Test
+    public void replace2AddsValueToBothMapsIfReplacedInExisting() {
+        SessionId key1 = new SessionId("key1");
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(key1, "value1");
+        }};
+        ConcurrentMap transition = new ConcurrentHashMap() {{
+            put(new SessionId("key2"), "value2");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        assertThat(transitionStore.replace(key1, "value3")).isEqualTo("value1");
+        assertThat(existing.get(key1)).isEqualTo("value3");
+        assertThat(transition.get(key1)).isEqualTo("value3");
+    }
+
+    @Test
+    public void replace2AddsValueToNeitherMapIfNotReplacedInExisting() {
+        SessionId key1 = new SessionId("key1");
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(key1, "value1");
+        }};
+        ConcurrentMap transition = new ConcurrentHashMap() {{
+            put(new SessionId("key2"), "value2");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        SessionId key3 = new SessionId("key3");
+        assertThat(transitionStore.replace(key3, "value3")).isEqualTo(null);
+        assertThat(existing.containsKey(key3)).isFalse();
+        assertThat(transition.containsKey(key3)).isFalse();
+    }
+
+    @Test
+    public void replace2DoesSameAsReplace2WhenUsingSingleMap() {
+        SessionId key1 = new SessionId("key1");
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(key1, "value1");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.empty());
+        assertThat(transitionStore.replace(key1, "value3")).isEqualTo("value1");
+        assertThat(existing.get(key1)).isEqualTo("value3");
+    }
+
+    @Test
+    public void transitionSizeShouldReturn0WhenNoMap() {
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(new SessionId("key1"), "value1");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.empty());
+        assertThat(transitionStore.transitionSize()).isEqualTo(0);
+    }
+
+    @Test
+    public void transitionSizeShouldReturnSizeOfTransitionMap() {
+        ConcurrentMap existing = new ConcurrentHashMap() {{
+            put(new SessionId("key1"), "value1");
+        }};
+        ConcurrentMap transition = new ConcurrentHashMap() {{
+            put(new SessionId("key2"), "value2");
+            put(new SessionId("key3"), "value3");
+        }};
+
+        TransitionStore transitionStore = new TransitionStore(existing, Optional.of(transition));
+        assertThat(transitionStore.transitionSize()).isEqualTo(transition.size());
+    }
+
+    @Test
+    public void getDifferingKeysShouldGiveSessionIdsOnlyInExistingMap() {
+        SessionId key1 = new SessionId("key1");
```

Comment:
getDifferingSessionIds() will return empty if both existing and transition have same session ids but these session ids have different values. Can we add a test case to test this scenario?